core/domain/exp_services.py

```diff
@@ -1270,3 +1273,44 @@ def search_explorations(query, limit, sort=None, cursor=None):
     """
     return search_services.search(
         query, SEARCH_INDEX_EXPLORATIONS, cursor, limit, sort, ids_only=True)
+
+         
+def _is_suggestion_valid(thread_id, exploration_id): 
+    """Check if the suggestion is still valid. A suggestion is considered 
+    invalid if the exploration version has updated since the suggestion was 
+    made.""" 
+ 
+    states = exp_models.ExplorationModel.get(id=exploration_id).states 
+    suggestion = (feedback_models.SuggestionModel 
+        .get_by_exploration_and_thread_id(exploration_id, thread_id)) 
+    if (suggestion.state_name in states.keys() and 
+           suggestion.state_content == states.get(suggestion.state_name)): 
+        return True 
+    return False        
+
+
+def accept_suggestion(editor_id, change_list, thread_id, exploration_id, 
+                      message):
+    """If the suggestion is valid, accepts it by updating the exploration.
+    Raises an exception if the exploration is not valid."""
+
+    if _is_suggestion_valid(thread_id, exploration_id):
+        commit_message = ('Accepted suggestion by %s: %s' % ( 
+            editor_id, message))
+        username = user_services.get_human_readable_user_ids([editor_id])[0]
+        update_exploration(
+            username, exploration_id, change_list, commit_message)
+        thread = (feedback_models.FeedbackThreadModel.
+            get_by_exp_and_thread_id(exploration_id, thread_id))
+        thread.status = feedback_models.STATUS_CHOICES_FIXED
+        thread.put()
+    else:
+        raise Exception('Suggestion Invalid')
+
+
+def reject_suggestion(thread_id, exploration_id):
+    """Set the state of a suggetion to REJECTED."""
+
+    thread = (feedback_models.FeedbackThreadModel.
+        get_by_exp_and_thread_id(exploration_id, thread_id))
+    thread.status = feedback_models.STATUS_CHOICES_IGNORED
```

Comment:
I've added the put() but I'm not sure why it wasn't caught in the test. How is the fetch returning the update object when put() wasn't even called.